Flaky test: test_grpc_protocol/test.py::test_cancel_while_generating_output

_Important: This issue was automatically generated and is used by CI for matching failures. DO NOT modify the body content. DO NOT remove labels._

Test name: test_grpc_protocol/test.py::test_cancel_while_generating_output
Failure reason: AssertionError
CI report: [Integration tests (amd_tsan, 6/6)](https://s3.amazonaws.com/clickhouse-test-reports/json.html?PR=105334&sha=8006f97a75b8020900c12641809d69eaa153c14c&name_0=PR&name_1=Integration%20tests%20%28amd_tsan%2C%206%2F6%29)

Failing test history: [cidb](https://play.clickhouse.com#V0lUSAogICAgOTAgQVMgaW50ZXJ2YWxfZGF5cwpTRUxFQ1QKICAgIHRvU3RhcnRPZkRheShjaGVja19zdGFydF90aW1lKSBBUyBkYXksCiAgICBjb3VudCgpIEFTIGZhaWx1cmVzLAogICAgZ3JvdXBVbmlxQXJyYXkocHVsbF9yZXF1ZXN0X251bWJlcikgQVMgcHJzLAogICAgYW55KHJlcG9ydF91cmwpIEFTIHJlcG9ydF91cmwKRlJPTSBjaGVja3MKV0hFUkUgKG5vdygpIC0gdG9JbnRlcnZhbERheShpbnRlcnZhbF9kYXlzKSkgPD0gY2hlY2tfc3RhcnRfdGltZQogICAgQU5EIHRlc3RfbmFtZSA9ICd0ZXN0X2dycGNfcHJvdG9jb2wvdGVzdC5weTo6dGVzdF9jYW5jZWxfd2hpbGVfZ2VuZXJhdGluZ19vdXRwdXQnCiAgICAtLSBBTkQgY2hlY2tfbmFtZSA9ICdJbnRlZ3JhdGlvbiB0ZXN0cyAoYW1kX3RzYW4sIDYvNiknCiAgICBBTkQgdGVzdF9zdGF0dXMgSU4gKCdGQUlMJywgJ0VSUk9SJykKICAgIEFORCAocHVsbF9yZXF1ZXN0X251bWJlciA9IDAgT1IgYmFzZV9yZWYgSU4gKCdtYXN0ZXInKSkKICAgIEFORCB0ZXN0X2NvbnRleHRfcmF3IExJS0UgJyVBc3NlcnRpb25FcnJvciUnCkdST1VQIEJZIGRheQpPUkRFUiBCWSBkYXkgREVTQwo=)

Test output:
```
File: test_grpc_protocol/test.py:607 - in test_cancel_while_generating_output
    assert output == b"0\t0\n1\t0\n2\t0\n3\t0\n"
E   AssertionError: assert b'0\t0\n1\t0\n' == b'0\t0\n1\t0\n2\t0\n3\t0\n'
E     
E     Full diff:
E     - (b'0\t0\n1\t0\n2\t0\n3\t0\n')
E     + (b'0\t0\n1\t0\n')
```

<!-- ch-version-info:start -->
### Version info
- Resolved by: #105415
- Merged into: `26.5.1.917`
<!-- ch-version-info:end -->

--- a/tests/integration/test_grpc_protocol/test.py
+++ b/tests/integration/test_grpc_protocol/test.py
@@ -604,7 +604,13 @@ def test_cancel_while_generating_output():
     output = b""
     for result in results:
         output += result.output
-    assert output == b"0\t0\n1\t0\n2\t0\n3\t0\n"
+    # The exact number of rows emitted before the cancel takes effect depends on
+    # how the server-side block production races against the cancel signal,
+    # which is timing-sensitive under load. Verify cancellation interrupted the
+    # query mid-stream by checking the output is a strict prefix of the full result.
+    full_output = b"".join(b"%d\t0\n" % i for i in range(10))
+    assert full_output.startswith(output), f"output not a prefix of full result: {output!r}"
+    assert len(output) < len(full_output), "cancel did not interrupt: got the full result"
 
 
 def test_compressed_output():

